test/compress/issue-2871.js

```diff
@@ -0,0 +1,17 @@
+comparison_with_undefined: {
+    options = {
+        comparisons: true,
+    }
+    input: {
+        a == undefined
+        a != undefined
+        a === undefined
+        a !== undefined
```

Comment:
Just for completeness please repeat these four lines with:
* `undefined` on the left
* `void 0` on the left